cmd/juju/commands/bundle.go

```diff
@@ -138,3 +182,20 @@ func (h *bundleHandler) setAnnotations(id string, p bundlechanges.SetAnnotations
 	// TODO frankban: implement this method.
 	return nil
 }
+
+// checkCompatibleCharms ensures that the charms with the given ids are
+// compatible, meaning an upgrade from one to the other is allowed.
+func checkCompatibleCharms(id1, id2 string) error {
+	ref1, err := charm.ParseReference(id1)
+	if err != nil {
+		return errors.Annotatef(err, "cannot parse charm URL %q", id1)
+	}
+	ref2, err := charm.ParseReference(id2)
+	if err != nil {
+		return errors.Annotatef(err, "cannot parse charm URL %q", id2)
+	}
+	if (ref1.Name != ref2.Name) || (ref1.User != ref2.User) {
```

Comment:
If a charm is promulgated and we change the user namespace, what happens with this condition?